src/core/MOM_barotropic.F90

```diff
@@ -6437,15 +6438,21 @@ subroutine barotropic_init(u, v, h, Time, G, GV, US, param_file, diag, CS, &
     Mean_SL = G%Z_ref
     Z_to_H = GV%Z_to_H ; if (.not.GV%Boussinesq) Z_to_H = GV%RZ_to_H * CS%Rho_BT_lin
     do j=js,je ; do I=is-1,ie
-      if (G%OBCmaskCu(I,j) > 0.) then
-        CS%IDatu(I,j) = G%OBCmaskCu(I,j) * 2.0 / (Z_to_H * ((G%bathyT(i+1,j) + G%bathyT(i,j)) + 2.0*Mean_SL))
+      htot = max(G%bathyT(i+1,j) + G%Z_ref, 0.0) + max(G%bathyT(i,j) + G%Z_ref, 0.0)
+      if (G%OBCmaskCu(I,j) * htot > 0.) then
+        ! Using CS%IDatu(I,j) = G%OBCmaskCu(I,j) * 2.0 / htot would be better but it changes answer.
+        CS%IDatu(I,j) = G%OBCmaskCu(I,j) * 2.0 / (Z_to_H * (max(G%bathyT(i+1,j) + G%Z_ref, 0.0) &
+                                                          + max(G%bathyT(i,j) + G%Z_ref, 0.0)))
```

Comment:
The denominator could be expressed in terms of `htot` here?

Reply:
I agree. mistaken it  for a `* Ihtot` scenario. (even that it should be OK in this case, right? Since the numerator has 0 as its significand)